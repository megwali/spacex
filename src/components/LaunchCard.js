import moment from 'moment';

import defaultImage from '../assets/images/spacex_logo_square.png';
import {
  Card,
  Content,
  FailedStatus,
  LaunchDate,
  MissionDetails,
  MissionName,
  SuccessStatus,
  UpcomingStatus
} from './styledComponents';


const LaunchCard = ({ launch }) => {
  const { flight_number, launch_date_local, launch_success, links, mission_name, upcoming } = launch;
  const src = links?.mission_patch_small || defaultImage;

  return (
    <Card to={`/${flight_number}`}>
      <Content>
        <img alt={mission_name} height="56" width="56" src={src} />

        <MissionDetails>
          <MissionName>{mission_name}</MissionName>
          <LaunchDate>{moment(launch_date_local).format('MMM Do YYYY, h:mm:ss a')}</LaunchDate>
        </MissionDetails>
      </Content>
      <div>
        {upcoming && <UpcomingStatus>Upcoming</UpcomingStatus>}
        {launch_success && <SuccessStatus>Successful</SuccessStatus>}
        {!upcoming && !launch_success && <FailedStatus>Failed</FailedStatus>}
      </div>
    </Card>
  );
};

export default LaunchCard;
