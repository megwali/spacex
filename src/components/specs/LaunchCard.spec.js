import { create } from 'react-test-renderer';
import LaunchCard from '../LaunchCard';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('LaunchCard', () => {
  it ('renders', () => {
    const component = create(<LaunchCard launch={{ flight_number: 1, launch_date_local: '2020-08-30T19:18:00-04:00' }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
