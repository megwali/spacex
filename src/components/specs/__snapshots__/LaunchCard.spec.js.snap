// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchCard renders 1`] = `
<Link
  className="sc-bdfBwQ CdKhw"
  to="/1"
>
  <div
    className="sc-gsTCUz fOxzTe"
  >
    <img
      height="56"
      src="spacex_logo_square.png"
      width="56"
    />
    <div
      className="sc-hKgILt cVPoMr"
    >
      <p
        className="sc-eCssSg liPRvx"
      />
      <p
        className="sc-dlfnbm NlWAF"
      >
        Aug 31st 2020, 12:18:00 am
      </p>
    </div>
  </div>
  <div>
    <p
      className="sc-jSgupP sc-gKsewC bPHQOa eOGKIx"
    >
      Failed
    </p>
  </div>
</Link>
`;
